lib/Sema/CSSimplify.cpp

```diff
@@ -9211,13 +9206,16 @@ bool ConstraintSystem::recordFix(ConstraintFix *fix, unsigned impact) {
 
   if (isAugmentingFix(fix)) {
     Fixes.push_back(fix);
-  } else {
+  } else if (auto *anchor = getAsExpr(fix->getAnchor())) {
```

Comment:
Should we assert that a non-augmenting fix has an expr anchor? Otherwise we'll silently drop it here

Reply:
Actually I think I'll just refactor this code to avoid depending on anchor being an expression.

Reply:
Done, @hamishknight how does that look?

Reply:
Looks great, thanks!